modules/ting_das/includes/ting_das.client.inc

```diff
@@ -0,0 +1,430 @@
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setIssueOfComponent(string $issueOfComponent) {
+    $this->issueOfComponent = $issueOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get article issue number.
+   *
+   * @return string
+   *    The issue number.
+   */
+  public function getIssueOfComponent() {
+    return $this->issueOfComponent;
+  }
+
+  /**
+   * Set the the volume number.
+   *
+   * @param string $volumeOfComponent
+   *   The article volume number.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setVolumeOfComponent(string $volumeOfComponent) {
+    $this->volumeOfComponent = $volumeOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get the article volume number.
+   *
+   * @return string|null
+   *   The article volume number.
+   */
+  public function getVolumeOfComponent() {
+    return $this->volumeOfComponent;
+  }
+
+  /**
+   * Set the page number in the publication.
+   *
+   * @param string $pagesOfComponent
+   *   The page number.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setPagesOfComponent(string $pagesOfComponent) {
+    $this->pagesOfComponent = $pagesOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get the page number.
+   *
+   * @return string|null
+   *   The page number.
+   */
+  public function getPagesOfComponent() {
+    return $this->pagesOfComponent;
+  }
+
+  /**
+   * Send request to the service.
+   *
+   * @return bool
+   *   TRUE if request was send else FALSE.
+   */
+  public function send() {
+    $options = array(
+      'headers' => array(
+        'Content-Type' => 'application/xml; charset=UTF8',
+      ),
+      'body' => $this->buildMessage(),
+    );
+
+    try {
+      $client = new Client();
+      $response = $client->request('POST', $this->url, $options);
+
+      if ($response->getStatusCode() == 204) {
+        watchdog('ting_das', 'Service returned: !code', array('!code' => $response->getStatusCode()), WATCHDOG_ERROR);
+        return TRUE;
+      }
+    }
+    catch (GuzzleException $exception) {
+      watchdog_exception('ting_das', $exception);
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Build XML message.
+   *
+   * @return string
+   *   The message as XML.
+   */
+  private function buildMessage() {
+    $xml = new SimpleXMLElement('<?xml version="1.0" encoding="utf-8"?><placeCopyRequest xmlns="http://statsbiblioteket.dk/xws/elba-placecopyrequest-schema"/>');
+
+    // Set required fields.
+    $xml->addChild('ws_user', $this->wsUsername);
+    $xml->addChild('ws_password', $this->wsPassword);
+    $xml->addChild('pid', $this->getPid());
+    $xml->addChild('user_loaner_id', $this->getUserLoanerId());
+    $xml->addChild('userName', $this->getUserName());
+    $xml->addChild('userMail', $this->getUserMail());
+    $xml->addChild('user_interest_date', $this->getUserInterestDate('Y-m-d'));
```

Comment:
Snake + camel casing are all over the place 🙄 .

Reply:
That's the DAS service not something that related to this PR.